widget/ui/src/components/RefreshProgressButton/RefreshProgressButton.tsx

```diff
@@ -0,0 +1,66 @@
+/* eslint-disable @typescript-eslint/no-magic-numbers */
+import type { SvgIconPropsWithChildren } from '../SvgIcon';
+
+import React, { createElement } from 'react';
+
+import { SvgIcon } from '../SvgIcon';
+
+const Ellipse = (props: { fill?: string; progress?: number }) => {
+  return (
+    <ellipse
+      cx="12.5"
+      cy="12.5"
+      rx="7.7"
+      ry="7.7"
+      strokeWidth="2"
+      stroke={props.fill || 'currentColor'}
+      strokeLinecap="round"
+      fill="none"
+      strokeDasharray={100}
+      strokeDashoffset={props.progress || 59}
+      style={{
+        transform: 'rotate(20deg)',
+        transformOrigin: 'center',
+      }}
+    />
+  );
+};
+function RefreshProgressButton(
+  props: SvgIconPropsWithChildren & { progress: any }
+) {
+  const normalizedProgress = Math.min(props.progress / 2, 41);
+  const opacity =
+    normalizedProgress !== 41 ? 0 : normalizeValue(props.progress);
+
+  function normalizeValue(value: number): number {
+    return (value - 82) / 10;
+  }
```

Comment:
In this instance, I believe that employing a function expression enhances readability.

Reply:
fixed.